packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -281,6 +284,12 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
     }
   }
 
+  void _handleWebFirstFrame(Duration _) {
+    assert(kIsWeb);
+    const MethodChannel methodChannel = MethodChannel('flutter/service_worker');
+    methodChannel.invokeMethod<void>('first-frame');
```

Comment:
btw, you are not checking for the method name in the engine.

Reply:
I don't think it is important unless we expand the number of actions here. i.e. building in service worker auto-update dialog or something.